indico/modules/events/persons/controllers.py

```diff
@@ -153,3 +154,58 @@ def _find_event_persons(self, person_ids):
         return (self.event_new.persons
                 .filter(EventPerson.id.in_(person_ids), EventPerson.email != '')
                 .all())
+
+
+class RHGrantSubmissionRights(RHConferenceModifBase):
+    """Grants submission rights to all contribution speakers"""
+
+    def _process(self):
+        for cont in self._target.as_event.contributions:
+            speakers = cont.speakers[:]
+            for subcontrib in cont.subcontributions:
+                speakers += subcontrib.speakers
+            for speaker in speakers:
+                principal = speaker.person.principal
+                if principal:
+                    cont.update_principal(principal, add_roles={'submit'})
+                    self.event_new.log(EventLogRealm.management, EventLogKind.positive, 'Protection',
```

Comment:
yes please. we could probably include the detailed list of people in the `data` of the log entry if needed